src/engine/strat_engine/tests/util.rs

```diff
@@ -9,43 +9,72 @@ use devicemapper::{DM, DmFlags, DevId};
 use nix::mount::{MNT_DETACH, umount2};
 use mnt::get_submounts;
 
+
+mod cleanup_errors {
+    error_chain!{}
+}
+
+use self::cleanup_errors::*;
+
+
 /// Attempt to remove all device mapper devices which match the stratis naming convention.
 /// FIXME: Current implementation complicated by https://bugzilla.redhat.com/show_bug.cgi?id=1506287
-fn dm_stratis_devices_remove() {
+fn dm_stratis_devices_remove() -> Result<()> {
 
     let dm = DM::new().unwrap();
 
+    let mut remain = Vec::new();
     loop {
         let mut progress_made = false;
-        for d in dm.list_devices()
-                .unwrap()
+
+        let device_list = dm.list_devices()
+            .chain_err(|| "Unable to retrieve dm devices")?;
+        for d in device_list
                 .iter()
                 .filter(|d| format!("{}", d.0.as_ref()).starts_with("stratis-1")) {
-            progress_made |= dm.device_remove(&DevId::Name(&d.0), DmFlags::empty())
-                .is_ok();
+
+            match dm.device_remove(&DevId::Name(&d.0), DmFlags::empty()) {
+                Ok(_) => progress_made = true,
+                Err(_) => remain.push(format!("{}", d.0.as_ref())),
+            }
         }
 
         if !progress_made {
+            if remain.len() != 0 {
+                bail!("We were unable to remove all stratis device mapper devices {:?}",
+                      remain);
+            }
             break;
+        } else {
+            remain.clear();
         }
     }
+
+    Ok(())
 }
 
 /// Try and un-mount any filesystems that have the name stratis in the mount point.
-fn stratis_filesystems_unmount() {
-    for m in get_submounts(&PathBuf::from("/"))
-            .unwrap()
+fn stratis_filesystems_unmount() -> Result<()> {
+
+    let mounts = get_submounts(&PathBuf::from("/"))
```

Comment:
For the entire body of this method, I would consider doing something more like:

```
|| {
  for m in get_submounts(&PathBuf::from("/"))?.iter().filter(...) {
    umount2(...)?; 
  }
}().map_err(|e| Error::with_chain(e, "Failed to unmount all mounted filesystems belonging to stratis")
```

or a fancier version that was able to report the unmounted filesystems.

The benefit is that you're able to chain onto the other errors a higher level error which describes the way in which this method, the one in which you're doing the chaining, failed. I would consider trying something similar in the more complicate method above, as well, but there's a lot more going on there, so harder to explain.